### ISSUE TITLE
Add support for keras.losses.Loss reduction

### ISSUE THREAD

**Issue by owenvallis** [tensorflow/similarity#200]
Currently the losses call reduce mean and return the final loss scaler. We should refactor the loss functions to return the per example loss and then perform the loss reduction using the reduction param. This will provide proper support for computing the loss when using distributed training strategies.

**Comment by owenvallis** [comment]
Updated all losses to use Loss reduction and include the name and reduction args in the from_config. e90ec5d239f9cb0b5d40ae37a599f23485ed48f1
